File: solutions/069.Sqrt(x)/AC_Newton_logn.py
# Author:      illuz <iilluzen[at]gmail.com>
# File:        AC_Newton_logn.py
# Create Date: 2015-02-06 23:33:33
# Descripton:   

class Solution:
    # @param x, an integer
    # @return an integer
    def sqrt(self, x):
        if x == 0:
            return 0
        lasty, y = 0.0, 1.0
        while y != lasty:
            lasty = y
            y = (y + x / y) / 2.0
        return int(y)

s = Solution()
print s.sqrt(3)

File: solutions/069.Sqrt(x)/AC_Newton_logn.cpp
/*
*  Author:      illuz <iilluzen[at]gmail.com>
*  File:        AC_Newton_logn.cpp
*  Create Date: 2015-02-06 23:25:45
*  Descripton:  Newton's method.
*/

#include <bits/stdc++.h>

using namespace std;
const int N = 0;

class Solution {
public:
    int sqrt(int x) {
        if (x == 0)
            return 0;
        double lasty = 0;
        double y = 1;
        while (y != lasty) {
            lasty = y;
            y = (y + x / y) / 2;
        }
        return int(y);
    }
};

int main() {
    int x;
    Solution s;
    while (cin >> x)
        cout << s.sqrt(x) << endl;
    return 0;
}

